experimental/LieAlgebras/test/CartanMatrix-test.jl

```diff
@@ -0,0 +1,154 @@
+    @test cartan_symmetrizer(cartan_matrix(:G, 2)) == [1, 3]
+
+    @test cartan_symmetrizer(ZZ[2 -2 0; -1 2 -1; 0 -1 2]) == [1, 2, 2]
+    @test cartan_symmetrizer(ZZ[2 0 -1 0; 0 2 0 -2; -2 0 2 0; 0 -1 0 2]) == [2, 1, 1, 2]
+
+    # affine type
+    @test cartan_symmetrizer(ZZ[2 -2; -2 2]) == [1, 1] # A1~1
+    @test cartan_symmetrizer(ZZ[2 -2 0; -1 2 -1; 0 -2 2]) == [1, 2, 1] # B3~1
+    @test cartan_symmetrizer(ZZ[2 -4; -1 2]) == [1, 4] # A1~2
+
+    # hyperbolic type
+    @test cartan_symmetrizer(ZZ[2 -2; -3 2]) == [3, 2]
+  end
+
+  @testset "cartan_bilinear_form(gcm::ZZMatrix; check::Bool)" begin
+    # finite type
+    @test cartan_bilinear_form(cartan_matrix(:A, 2)) == ZZ[2 -1; -1 2]
+    @test cartan_bilinear_form(cartan_matrix(:B, 4)) ==
+      ZZ[4 -2 0 0; -2 4 -2 0; 0 -2 4 -2; 0 0 -2 2]
+    @test cartan_bilinear_form(cartan_matrix(:C, 4)) ==
+      ZZ[2 -1 0 0; -1 2 -1 0; 0 -1 2 -2; 0 0 -2 4]
+    @test cartan_bilinear_form(cartan_matrix(:F, 4)) ==
+      ZZ[4 -2 0 0; -2 4 -2 0; 0 -2 2 -1; 0 0 -1 2]
+    @test cartan_bilinear_form(cartan_matrix(:G, 2)) == ZZ[2 -3; -3 6]
+
+    @test cartan_bilinear_form(cartan_matrix((:B, 2), (:A, 1))) == ZZ[4 -2 0; -2 2 0; 0 0 2]
+    @test cartan_bilinear_form(ZZ[2 0 -1 0; 0 2 0 -2; -2 0 2 0; 0 -1 0 2]) ==
+      ZZ[4 0 -2 0; 0 2 0 -2; -2 0 2 0; 0 -2 0 4]
+
+    # affine type
+    @test cartan_bilinear_form(ZZ[2 -2; -2 2]) == ZZ[2 -2; -2 2]
+    @test cartan_bilinear_form(ZZ[2 -2 0; -1 2 -1; 0 -2 2]) == ZZ[2 -2 0; -2 4 -2; 0 -2 2]
+    @test cartan_bilinear_form(ZZ[2 -4; -1 2]) == ZZ[2 -4; -4 8]
+  end
+
+  @testset "cartan_type(gcm::ZZMatrix; check::Bool)" begin
+    # test if we follow our own conventions
+    @test cartan_type(cartan_matrix(:A, 3); check=false) == [(:A, 3)]
+    @test cartan_type(cartan_matrix(:B, 2); check=false) == [(:B, 2)]
+    @test cartan_type(cartan_matrix(:C, 2); check=false) == [(:C, 2)]
+
+    # tests for irreducibles
+    @test cartan_type(cartan_matrix(:A, 1); check=false) == [(:A, 1)]
+    @test cartan_type(cartan_matrix(:A, 2); check=false) == [(:A, 2)]
+
+    @test cartan_type(cartan_matrix(:B, 3); check=false) == [(:B, 3)]
+    @test cartan_type(ZZ[2 -2 0; -1 2 -1; 0 -1 2]; check=false) == [(:B, 3)]
+
+    @test cartan_type(cartan_matrix(:C, 3); check=false) == [(:C, 3)]
+    @test cartan_type(ZZ[2 -1 0; -2 2 -1; 0 -1 2]; check=false) == [(:C, 3)]
+
+    @test cartan_type(cartan_matrix(:D, 4); check=false) == [(:D, 4)]
+    @test cartan_type(cartan_matrix(:D, 6); check=false) == [(:D, 6)]
+    @test cartan_type(cartan_matrix(:E, 6); check=false) == [(:E, 6)]
+    @test cartan_type(cartan_matrix(:E, 7); check=false) == [(:E, 7)]
+    @test cartan_type(cartan_matrix(:E, 8); check=false) == [(:E, 8)]
+    @test cartan_type(cartan_matrix(:F, 4); check=false) == [(:F, 4)]
+    @test cartan_type(cartan_matrix(:G, 2); check=false) == [(:G, 2)]
+
+    # for F4 and G2 we also allow the transposed Cartan matrix
```

Comment:
please add transposed tests for B3 and C3 here

Reply:
This does not make sense. `transpose(cartan_matrix(:B, 3)) == cartan_matrix(:C, 3)`.

Reply:
True...